src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs

```diff
@@ -22,93 +23,147 @@
+                    }
 
-            var resourceName = Path.GetFileNameWithoutExtension(resourceFile.Path);
-            if (options.TryGetValue("build_metadata.AdditionalFiles.RelativeDir", out var relativeDir))
-            {
-                resourceName = relativeDir.Replace(Path.DirectorySeparatorChar, '.').Replace(Path.AltDirectorySeparatorChar, '.') + resourceName;
-            }
+                    if (!options.TryGetValue("build_metadata.AdditionalFiles.OmitGetResourceString", out var omitGetResourceStringText)
+                        || !bool.TryParse(omitGetResourceStringText, out var omitGetResourceString))
+                    {
+                        omitGetResourceString = false;
+                    }
 
-            if (!options.TryGetValue("build_metadata.AdditionalFiles.OmitGetResourceString", out var omitGetResourceStringText)
-                || !bool.TryParse(omitGetResourceStringText, out var omitGetResourceString))
-            {
-                omitGetResourceString = false;
-            }
+                    if (!options.TryGetValue("build_metadata.AdditionalFiles.AsConstants", out var asConstantsText)
+                        || !bool.TryParse(asConstantsText, out var asConstants))
+                    {
+                        asConstants = false;
+                    }
 
-            if (!options.TryGetValue("build_metadata.AdditionalFiles.AsConstants", out var asConstantsText)
-                || !bool.TryParse(asConstantsText, out var asConstants))
-            {
-                asConstants = false;
-            }
+                    if (!options.TryGetValue("build_metadata.AdditionalFiles.IncludeDefaultValues", out var includeDefaultValuesText)
+                        || !bool.TryParse(includeDefaultValuesText, out var includeDefaultValues))
+                    {
+                        includeDefaultValues = false;
+                    }
 
-            if (!options.TryGetValue("build_metadata.AdditionalFiles.IncludeDefaultValues", out var includeDefaultValuesText)
-                || !bool.TryParse(includeDefaultValuesText, out var includeDefaultValues))
-            {
-                includeDefaultValues = false;
-            }
+                    if (!options.TryGetValue("build_metadata.AdditionalFiles.EmitFormatMethods", out var emitFormatMethodsText)
+                        || !bool.TryParse(emitFormatMethodsText, out var emitFormatMethods))
+                    {
+                        emitFormatMethods = false;
+                    }
 
-            if (!options.TryGetValue("build_metadata.AdditionalFiles.EmitFormatMethods", out var emitFormatMethodsText)
-                || !bool.TryParse(emitFormatMethodsText, out var emitFormatMethods))
-            {
-                emitFormatMethods = false;
-            }
+                    return new[]
```

Comment:
This concerns me. Arrays are unlikely to compare equal across different runs. Do we need an equatable array implementation? e.g, https://github.com/CommunityToolkit/dotnet/blob/main/src/CommunityToolkit.Mvvm.SourceGenerators/Helpers/EquatableArray%7BT%7D.cs

Reply:
This is returning from `SelectMany`, so the array elements are going to be expanded by the caller.